### PR TITLE
Fix address filtering for /ip6, add tests

### PR DESCRIPTION
**Pull request by lgierth**
License: MIT
Signed-off-by: Lars Gierth <larsg@systemli.org>

### CODE REVIEW

**Comment by lgierth** [comment]
Fixed the tests

File: p2p/net/filter/filter.go [hunk 1]
```diff
@@ -27,18 +26,24 @@ func (fs *Filters) AddDialFilter(f *net.IPNet) {
...
+		return false
+	}
+	netaddr, err := manet.ToNetAddr(maddr[0])
 	if err != nil {
 		// if we cant parse it, its probably not blocked
 		return false
 	}
+	netip := net.ParseIP(netaddr.String())
+	if netip == nil {
+		return false
```
Comment (jbenet) [inline p2p/net/filter/filter.go]: right now these filters are ip-specific. in the future we may want to express filters of other kinds (like tcp ports, etc)

**Comment by jbenet** [comment]
LGTM. @whyrusleeping ?

**Comment by whyrusleeping** [comment]
yeah, LGTM

**Comment by Kubuxu** [comment]
Thanks for that.

**Pull request merged**